lib/system/dollars.nim

```diff
@@ -160,3 +165,27 @@ proc `$`*[T](x: openArray[T]): string =
   ## .. code-block:: Nim
   ##   $(@[23, 45].toOpenArray(0, 1)) == "[23, 45]"
   collectionToString(x, "[", ", ", "]")
+
+proc `$`*[T: ref | ptr](arg: T): string =
+  if arg == nil:
+    "nil"
+  else:
+    $arg[]
```

Comment:
There should be *some* difference between `echo somePtrToInt` and `echo someInt`

Reply:
For `ptr someType` maybe print the address and the object it points to

Reply:
you mean you would like to have an indirection sign like ``"-> 1"`` instead of ``"1"``?

Reply:
ok, my github was slow I didn't see your second message. I actually thought about printing the address, but honestly most of the time you really do not care about an address. Addresses are long and rarely what you really care about. That is why I omitted it. I also didn't indicate that the object is a pointer, because for me being a pointer is part of the type, and types aren't printed in string representation.

Reply:
> because for me being a pointer is part of the type, and types aren't printed in string representation.

I agree on this point

>  I actually thought about printing the address, but honestly most of the time you really do not care about an address.

i disagree on this point, with reference types the address is what matters most since equality (and hence identity) is base on that instead of its dereferenced content. Eg if you want to know whether 2 nodes are equal in a graph. This relates to my point in https://github.com/nim-lang/Nim/pull/13687#issuecomment-601112539

so for top-level types you'd print:
```
type Foo = ref object
  x: int
  f2: Foo
doAssert $Foo(x: 2) == 0xdeadbeef->(x: 2, f2: 0xdead0000)
```
analog to how `repr` does

Reply:
well for ``pointer`` I agree, the address is what should be printed and that is also what I do print here in this PR. But for everything else, pointer addresses are just visual noise, nothing else. And if you do care about an address you can still do: ``$cast[pointer](x)`` on it.